aptos-indexer-processors-sdk/sdk/src/common_steps/order_by_version_step.rs

```diff
@@ -0,0 +1,185 @@
+use crate::{
+    traits::{
+        pollable_async_step::PollableAsyncRunType, NamedStep, PollableAsyncStep, Processable,
+    },
+    types::transaction_context::TransactionContext,
+    utils::errors::ProcessorError,
+};
+use ahash::AHashMap;
+use anyhow::Result;
+use async_trait::async_trait;
+use std::time::Duration;
+
+pub struct OrderByVersionStep<Input>
+where
+    Self: Sized + Send + 'static,
+    Input: Send + 'static,
+{
+    pub ordered_versions: Vec<TransactionContext<Input>>,
+    pub unordered_versions: AHashMap<u64, TransactionContext<Input>>,
+    pub expected_next_version: u64,
+    // pub versions: BTreeSet<TransactionContext<Input>>,
+    // Duration to poll and return the ordered versions
+    pub poll_interval: Duration,
```

Comment:
out of scope: i think it would be nice if we can adjust this dynamically based on the current load or the number of unordered versions stored. For example, if gaps are frequently encountered, you might want to decrease the polling interval to reduce idle time.  if transactions arrive in order, you could incrase the interval to reduce overhead.

Reply:
We could also make another version of this step where it uses `max_ordered_versions` instead of `poll_interval`, where if `max_ordered_versions` is reached, then return them as output. It might be more helpful for Parquet.

Reply:
yeah that makes sense, potential enhancements we can include is combining this with polling or tuning `max_ordered_versions` based on the batch size optimal for Parquet.